vm/vm_test.go

```diff
@@ -409,15 +436,28 @@ func TestVM(t *testing.T) {
 						t.Fatal(err)
 					}
 				}
-				err = vm.Execute()
-				require.Equal(t, testCase.err, err)
+				return vm.Execute()
 			}
 			if len(splitSet) == 0 {
-				verify(nil)
+				err := verify(nil)
+				require.Equal(t, testCase.err, err)
 				return
 			}
-			for _, splitAsset := range splitSet {
-				verify(splitAsset)
+			// For splits, sort by ascending value so that we fail
+			// early on invalid zero-value locators.
+			splitAssets := maps.Values(splitSet)
+			sort.Slice(splitAssets, func(i, j int) bool {
+				return splitAssets[i].Asset.Amount <
+					splitAssets[j].Asset.Amount
+			})
+			for _, splitAsset := range splitAssets {
```

Comment:
Is this sorting actually required? Or was it that we weren't actually capturing the error value here to asserting based on that?

Reply:
Before, all the cases with expected errors didn't overlap with validating a split; the issue is that before we were requiring the same error value for each `vm.Execute()` call, but in practice we exit after the first non-nil err.

Here we sort so that, for the new test cases where we expect to fail on the zero-value root asset, those assets are validated first. Without sorting, the test case fails when a `vm.Execute()` call succeeds, before we ever get to the expected failure.